### ISSUE TITLE
M16 matom model failing in cdf_gen_from_array

### ISSUE THREAD

**Issue by jhmatthews** [sirocco-rt/sirocco#300]
This is clearly linked to LT2 and other issues such as #297 #293 

Code state: d04e57243b35c56bc749c4775fb6f92d95e873c7 - unmerged, currently open pull request #299 - but more or less identical to dev py82 (https://github.com/agnwinds/python/commit/b7f3abf0e8da00f44536a6e69b0290eab4a93e55)
Parameter file: [m16_h10he_80.pf.txt](https://github.com/agnwinds/python/files/1302363/m16_h10he_80.pf.txt)
Atomic data: recently updated version, pull from agnwinds/data. 

When I run above following parameter file, which is my AGN model with some updated data I get the following error in ionization cycle 10. 

```
!!python: Total photon luminosity before transphot 3.025925592089e+46

Photon       0 of 6666666 or  0.000 per cent 
Error: cdf_gen_from_array - only one point in supplied PDF
```

This is occurring in the photon cycles, rather than in photon generation or anything where we often use CDFs, which slightly surprised me. I have traced the error to a call to `one_ff()` from the k-packet machinery in `kpkt()`. This call is required when a k-packet is converted to an r-packet by free-free emission. When it does so, it supplies band boundaries and attempts to generate a frequency for the r-packet. 

More to follow...

**Comment by jhmatthews** [comment]
Ok, so here's why this happens. When we call one_ff from kpkt() we choose the new photon frequency by sampling a CDF of the free-free emissivity for that cell. But, if you look here: #187 - I changed the hardwired limits to be a huge range (1e14 and 1e20) so that one could in theory generate photons at any frequency.

The problem with this approach I adopted is that it's rubbish. By way of example, take the free-free emissivity in the cell that causes the above error. It looks like this: 
![free_free_emiss_in_cell](https://user-images.githubusercontent.com/3007249/30422684-975516f6-9938-11e7-8b9e-05b7d0493536.png)

The vertical lines mark the boundaries introduced by one_ff when you supply frequency limits of 1e14 and 1e20. the temp in this cell is 6000K, so because of the exp(-h nu / kT) factor you generate all your luminosity in the first bin and the CDF only has one point in it, and cdf_gen_from_array is now written so that the program quits when this happens.

This does of course mean that the free-free spectrum *in the ionization cycles* was probably way off in my macro-atom models. It will be interesting to know if this affects anything.

There are really three issues/questions here I think:
* Free free photon generation from kpkt() is clearly very bad as it will often default to some low frequency in cool cells. I think the frequency limit should thus be chosen as something like freqmax = (100.0) * kT / h for each cell. 
* in the normal wind generationI think must be ok as we use stratified sampling there, but worth checking if there's a better approach
* Should a CDF work if we only supply it one point? I would argue that it perhaps should, but on the other hand I would never have spotted this rather glaring error if the program didn't quit.
